### ISSUE TITLE
Import Broken

### ISSUE THREAD

**Issue by eldritchgreene** [simjanos-dev/LinguaCafe#309]
Thanks for the incredible app. I've run into a problem importing text. It's really bizarre but a specific Japanese text import corrupts an entire Book. I was making a book with 'songs' as chapters, when I imported a specific song lyric it pulled an error, tried it a couple times but eventually gave up and noticed it had frozen my Book into loading mode and the chapters were all gone. I've since deleted it and tried again, and every time the same text creates the same error, there is nothing weird in it, I removed the ellipses, a dot in the title (not shared), white spaces, but I can't pinpoint what the problem is. I think it may be a Kanji that is causing this? This is the text:

```
眠るような陽を浴びて 君はブロンズ色

南向きのベランダで
海を眺めている

やわらかな前髪のカール
憂いがちな眼を隠す
風はペパーミント
ブルーのソーダが 指先に揺れている
斜め横の椅子を選ぶのは
この角度からの君が とても綺麗だから

黙りこんだ貝殻が 深みできらめくよ
そう 大事なこと ぼくはまだ
話し忘れてたよ

沖をゆく客船の汽笛 旅に誘っても
ここを動かない
風はペパーミント
想い出の日々が グラスからはじけてる
旧い歌の低いハミングに
口笛でハーモニー 重なる音が溶けて消える

波は時を砂に変え
寄せる･･･やさしく･･･

そんな風にぼくたちも
愛せたらいいのに
水のように透明な
心ならいいのに

抱きしめた両手から逃げる
灼き すぎて 痛い わ って

風はペパーミント
ブルーのソーダが 指先に揺れている
空も海も遠のいてゆくよ
君のはにかんだ 笑顔だけを残して 
```

I've checked previous issues and there is at least one which resembles mine. I think a second one links to that one. I'm not as savvy as the posters but I managed to check one log and this is the error i found:

``[2024-06-12 12:49:15] local.ERROR: Undefined array key 237 {"userId":1,"exception":"[object] (ErrorException(code: 0): Undefined array key 237 at /var/www/html/app/Models/TextBlock.php:212)
[stacktrace]
#0 /var/www/html/app/Models/TextBlock.php(212): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->handleError(2, 'Undefined array...', '/var/www/html/a...', 212)
#1 /var/www/html/app/Services/ChapterService.php(226): App\\Models\\TextBlock->processTokenizedWords()
#2 /var/www/html/app/Http/Controllers/ChapterController.php(104): App\\Services\\ChapterService->updateChapter(1, 48, 'Test', '\\xE7\\x9C\\xA0\\xE3\\x82\\x8B\\xE3\\x82\\x88\\xE3\\x81\\x86\\xE3\\x81\\xAA...')
#3 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/Controller.php(54): App\\Http\\Controllers\\ChapterController->updateChapter(Object(App\\Http\\Requests\\Chapter\\UpdateChapterRequest))
#4 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php(45): Illuminate\\Routing\\Controller->callAction('updateChapter', Array)
#5 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/Route.php(261): Illuminate\\Routing\\ControllerDispatcher->dispatch(Object(Illuminate\\Routing\\Route), Object(App\\Http\\Controllers\\ChapterController), 'updateChapter')
#6 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/Route.php(205): Illuminate\\Routing\\Route->runController()
#7 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/Router.php(721): Illuminate\\Routing\\Route->run()
#8 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(128): Illuminate\\Routing\\Router->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))``

Thanks again.

**Comment by etherealite** [comment]
I copy pasted the input text [eldritchgreene](https://github.com/eldritchgreene) and it blew up in the exact same way for me.

**Comment by simjanos-dev** [comment]
Thank you for the bug report! 

> I copy pasted the input text [eldritchgreene](https://github.com/eldritchgreene) and it blew up in the exact same way for me.

That's good, I was worried it corrupted the database. I'll check it out later today probably.

**Comment by simjanos-dev** [comment]
It was a bug in my Japanese text processing script. I fixed it, it will be in v0.13. In the meantime, you can add a `。` character to the end of your text, it should fix it.

**Comment by eldritchgreene** [comment]
That's a relief, thanks so much. :)
